src/basic/json.h

```diff
@@ -0,0 +1,273 @@
+/* SPDX-License-Identifier: LGPL-2.1+ */
+
+#pragma once
+
+#include <stdbool.h>
+#include <stddef.h>
+#include <stdint.h>
+
+#include "macro.h"
+#include "string-util.h"
+#include "util.h"
+
+/*
+  In case you wonder why we have our own JSON implementation, here are a couple of reasons why this implementation has
+  benefits over various other implementatins:
+
+  - We need support for 64bit signed and unsigned integers, i.e. the full 64,5bit range of -9223372036854775808…18446744073709551615
+  - All our variants are immutable after creation
+  - Special values such as true, false, zero, null, empty strings, empty array, empty objects require zero memory
```

Comment:
"zero" is probably wrong...

Reply:
not, it's actually correct. If you parse the json "0" we won't allocate memory for that, but return `(void*) 4`, i.e. we encode in the returned ptr itself that this is a pointer. It's a nice little hack to suppress memory allocations for common special cases...

Reply:
Exactly ;). So it requires `sizeof(*)` bytes, but not 0. I know what you mean, but I don't want some troll to see this comment.

Reply:
will change to "zero memory" → "zero dynamic memory"